src/itensornetwork.jl

```diff
@@ -86,15 +102,33 @@ function insert_links(ψ::ITensorNetwork, edges::Vector=edges(ψ); cutoff=1e-15)
   return ψ
 end
 
-function ITensorNetwork(is::IndsNetwork, initstate::Function)
+function ITensorNetwork(
+  ::Type{ElT}, is::IndsNetwork, states_map::Dictionary
+) where {ElT<:Number}
   ψ = ITensorNetwork(is)
   for v in vertices(ψ)
-    ψ[v] = state(initstate(v), only(is[v]))
+    ψ[v] = ITensors.convert_leaf_eltype(ElT, state(only(is[v]), states_map[v]))
```

Comment:
I guess `ITensors.state` should optionally take an element type, could you raise an issue about that in ITensors.jl?

In the meantime, you should be able to just use `convert_eltype` here since it is just an ITensor (`ITensors.convert_leaf_eltype` is more for nested data structures like `MPS`, `Vector{ITensor}`, `ITensorNetwork`, etc.).

Reply:
Issue raised, and I switched to `convert_eltype`.